src/lib/operators/validate.cpp

```diff
@@ -71,6 +80,23 @@ std::shared_ptr<const Table> Validate::_on_execute(std::shared_ptr<TransactionCo
   auto job_end_chunk_id = ChunkID{0};
   auto job_row_count = uint32_t{0};
 
+  // Checking visibility on chunk level is an optimization to reduce the work done during validation.
+  // Instead of checking each row for visibility, we determine the largest begin CommitID (max_begin_cid) for an entire
```

Comment:
Start by explaining the ratio behind it and why it is possible, not by explaining what steps are done

Reply:
Better now.